### PR TITLE
Fix local-storage transformations for uneven partitions

### PR DESCRIPTION
**Pull request by tbennun**
- [x]  LocalStorage/MapTiling uneven copy does not produce SymExpr

### CODE REVIEW

File: dace/subsets.py [hunk 1]
```diff
@@ -176,6 +176,30 @@ def size(self, for_codegen=False):
...
+            int_ceil = sp.Function('int_ceil')
+            return [
+                ts * int_ceil(
+                    ((iMax.expr if isinstance(iMax, symbolic.SymExpr) else
+                      iMax) + 1 - (iMin.expr if isinstance(
+                          iMin, symbolic.SymExpr) else iMin)),
+                    (step.expr
+                     if isinstance(step, symbolic.SymExpr) else step))
+                for (iMin, iMax, step), ts in zip(self.ranges, self.tile_sizes)
+            ]
```
Comment (tbennun) [inline dace/subsets.py]: @alexnick83 If no one is calling this mode, do we need it?

File: dace/codegen/targets/cpp.py [hunk 2]
```diff
@@ -97,8 +97,9 @@ def memlet_copy_to_absolute_strides(dispatcher,
     # Ignore vectorization flag is a hack to accommmodate FPGA behavior,
     # where the pointer type is changed to a vector type, and addresses
     # thus shouldn't take vectorization into account.
-    copy_shape = memlet.subset.size()
-    copy_shape = [symbolic.overapproximate(s) for s in copy_shape]
+    copy_shape = memlet.subset.size_exact()
+    # copy_shape = memlet.subset.size()
+    # copy_shape = [symbolic.overapproximate(s) for s in copy_shape]
```
Comment (tbennun) [inline dace/codegen/targets/cpp.py]:
```suggestion
```

File: dace/subsets.py (hunk 1 above)
Comment (tbennun) [reply dace/subsets.py]: Also, this is why we have `symcpp` in the codegen, it makes the conversion there.

Comment (alexnick83) [reply dace/subsets.py]: I actually don't know why this mode exists. I just copied it as it was in the size method. Is it used there anywhere?

Comment (tbennun) [reply dace/subsets.py]: Not sure. We can remove it then

Comment (alexnick83) [reply dace/subsets.py]: I removed the path. We can just re-add it if we find a use-case for it.

**Pull request merged**